tools/deploy-all

```diff
@@ -1,16 +1,38 @@
-#! /bin/bash -e
-
+#! /bin/bash
+#
 # Perform complete deployment of JupyterHub,  including:
 #
 # 1. Update kubeconfig
 # 2. Nested secrets decryption and Helm chart upgrade
-# 3. Fetch service proxy public IP
+# 3. Helm install / upgrade
+# 4. Fetch service proxy public IP
 #
 
+set -eu
+
 cd $JUPYTERHUB_DIR
 
-awsudo $ADMIN_ARN aws eks update-kubeconfig --name ${DEPLOYMENT_NAME} --region us-east-1 --role-arn arn:aws:iam::${ACCOUNT_ID}:role/jupyterhub-admin
+secrets_yaml=deployments/${DEPLOYMENT_NAME}/secrets/${ENVIRONMENT}.yaml
+tmp_decrypted=${JUPYTERHUB_DIR}/env.yaml.decrypted
+
+awsudo $ADMIN_ARN aws eks update-kubeconfig --name ${DEPLOYMENT_NAME} --region us-east-1 --role-arn $ADMIN_ARN
```

Comment:
I'm curious why `--role-name $ADMIN_ARN` is required here?  Also, otherwise you could use your `update-kubeconfig` alias.

Reply:
I'm hesitant to use aliases in the scripts too much because it gets a little byzantine;  I think aliases are probably not everyone's style so I was trying to keep those as optional conveniences people can just ignore if they want to.